src/Compiler/Driver/FxResolver.fs

```diff
@@ -897,58 +798,51 @@ type internal FxResolver
-            // Interactive processes read their own configuration to find the running tfm
 
-            let tfm =
-                if isInteractive then
-                    match tryGetRunningDotNetCoreTfm () with
-                    | Some tfm -> tfm
-                    | _ -> getRunningDotNetFrameworkTfm ()
-                else
-                    let sdkDir = tryGetSdkDir () |> replayWarnings
-
-                    match sdkDir with
-                    | Some dir ->
-                        let dotnetConfigFile = Path.Combine(dir, "dotnet.runtimeconfig.json")
-                        use stream = FileSystem.OpenFileForReadShim(dotnetConfigFile)
-                        let dotnetConfig = stream.ReadAllText()
-                        let pattern = "\"tfm\": \""
-
-                        let startPos =
-                            dotnetConfig.IndexOf(pattern, StringComparison.OrdinalIgnoreCase)
-                            + pattern.Length
-
-                        let endPos = dotnetConfig.IndexOf("\"", startPos)
-                        let tfm = dotnetConfig[startPos .. endPos - 1]
-                        //printfn "GetTfmAndRid, tfm = '%s'" tfm
-                        tfm
-                    | None ->
-                        match tryGetRunningDotNetCoreTfm () with
-                        | Some tfm -> tfm
-                        | _ -> getRunningDotNetFrameworkTfm ()
+            let runningTfm = tryGetRunningTfm
 
+            // Coreclr has mechanism for getting rid
+            //      System.Runtime.InteropServices.RuntimeInformation.RuntimeIdentifier
+            // On Desktop framework compile it using osplatform+processarch+
             // Computer valid dotnet-rids for this environment:
             //      https://docs.microsoft.com/en-us/dotnet/core/rid-catalog
             //
             // Where rid is: win, win-x64, win-x86, osx-x64, linux-x64 etc ...
             let runningRid =
-                let processArchitecture = RuntimeInformation.ProcessArchitecture
-
-                let baseRid =
-                    if RuntimeInformation.IsOSPlatform(OSPlatform.Windows) then
-                        "win"
-                    elif RuntimeInformation.IsOSPlatform(OSPlatform.OSX) then
-                        "osx"
+                let rid =
+                    if isRunningOnCoreClr then
+                        // Use reflection to get the value for rid
+                        let rinfoType: Type option =
+                            Option.ofObj (Type.GetType("System.Runtime.InteropServices.RuntimeInformation, System.Runtime.InteropServices.RuntimeInformation", false))
+
+                        let ridProperty: PropertyInfo option =
+                            match rinfoType with
+                            | None -> None
+                            | Some t -> Option.ofObj (t.GetProperty("RuntimeIdentifier", BindingFlags.Public  ||| BindingFlags.Static))
```

Comment:
one thing to note here is that this will give you the 'most specific' rid. for example, right now on my windows 11 box on .net 7 preview 6 I get a rid of "win10-x64", which may not be what you're expecting.

Reply:
@baronfel, that is exactly what we want.